src/connect.rs

```diff
@@ -164,23 +165,19 @@ fn handle_new_connection_res(
                 ref mut pending_reads,
                 ..
             } => {
-                let node_info = NodeInfo {
-                    peer_addr,
-                    peer_cert_der: peer_cert_der.clone(),
-                };
                 let peer = Peer::Node { node_info };
-
-                if let Err(e) = c.event_tx.send(Event::ConnectedTo { peer }) {
+                if let Err(e) = c.event_tx.send(Event::ConnectedTo { peer: peer.clone() }) {
                     println!("Could not fire event: {:?}", e);
                 }
 
                 for pending_read in pending_reads.drain(..) {
                     communicate::dispatch_wire_msg(
-                        peer_addr,
+                        peer.clone(),
```

Comment:
```suggestion
                        peer,
```

Reply:
Can't do that: `dispatch_wire_msg()` is called multiple times in a loop and `Peer` does not implement `Copy`.